### PR TITLE
Refactor features

### PR DESCRIPTION
**Pull request by michalvasko**

### CODE REVIEW

File: src/tree_schema.h [hunk 1]
```diff
@@ -118,40 +118,25 @@ struct ly_set;
...
- * changed feature. On the contrary, ::lys_feature_enable_force() and ::lys_feature_disable_force() ignore all the if-feature
- * limitations.
- *
- * The ::lysc_feature_value() and ::lys_feature_value() functions differ only by their parameters. The ::lysc_iffeature_value()
- * is used to evaluate (possibly complex in YANG 1.1) logical expression from if-feature statement.
- *
- * The list of features of a particular YANG module is available in ::lys_module.features.
- *
- * To get know, if a specific schema node is currently disabled or enable, the ::lysc_node_is_disabled() function can be used.
+ * The iterate over all features of a particular YANG module, use ::lysp_feature_next().
```
Comment (rkrejci) [inline src/tree_schema.h:130]: To iterate over ...

File: src/tree_schema.h [hunk 2]
```diff
@@ -1947,12 +1905,22 @@ LY_ERR lysc_iffeature_value(const struct lysc_iffeature *iff);
...
  * @brief Get the current status of the provided feature.
  *
- * @param[in] feature Compiled feature statement to examine.
+ * @param[in] feature Feature statement to examine.
  * @return LY_SUCCESS if feature is enabled,
  * @return LY_ENOT if feature is disabled,
  * @return LY_ERR in case of error (invalid argument)
  */
-LY_ERR lysc_feature_value(const struct lysc_feature *feature);
+LY_ERR lysp_feature_value(const struct lysp_feature *feature);
```
Comment (rkrejci) [inline src/tree_schema.h:1913]: Not sure if it is really needed when the value is now clearly set by `LYS_FENABLED` without anything else.

**Review by rkrejci — CHANGES_REQUESTED** [review]

**Comment by rkrejci** [comment]
One more note regarding `lys_set_implemented()`. Features in the imported modules are all disabled and there is no other way to enable the selected features except the moment the module is changed to the implemented module and compiled, i.e. when the `lys_set_implemented()`  is called. So, there should be the `features` parameter added also to this function.

File: src/context.h [hunk 3]
```diff
@@ -469,9 +469,11 @@ uint32_t ly_ctx_internal_module_count(const struct ly_ctx *ctx);
  * @param[in] ctx Context to add to.
  * @param[in] name Name of the module to load.
  * @param[in] revision Optional revision date of the module. If not specified, the latest revision is loaded.
+ * @param[in] features Optional array of features ended with NULL to be enabled if the module is being implemented.
```
[the remark this answers is not in the document]
Comment (michalvasko) [reply src/context.h:472]: Fixed.

File: src/schema_compile.h [hunk 4]
```diff
@@ -29,14 +29,18 @@
...
+#define LYS_COMPILE_RPC_MASK        LYS_CONFIG_MASK /**< mask for the internal RPC options */
+#define LYS_COMPILE_NOTIFICATION    0x04            /**< Internal option when compiling schema tree of Notification */
+
+#define LYS_COMPILE_GROUPING        0x08            /**< Compiling (validation) of a non-instantiated grouping.
+                                                      In this case not all the restrictions are checked since they can
+                                                      be valid only in the real placement of the grouping.
+                                                      TODO - what specifically is not done */
+#define LYS_COMPILE_DISABLED        0x10            /**< Compiling a disabled subtree (by its if-features). Meaning
+                                                      it will be removed at the end of compilation and should not be
+                                                      added to anyunres sets. */
```
[the remark this answers is not in the document]
Comment (michalvasko) [reply src/schema_compile.h]: Fixed.

File: src/tree_schema.h (hunk 1 above)
Comment (michalvasko) [reply src/tree_schema.h]: Fixed.

File: src/tree_schema.h (hunk 2 above)
Comment (michalvasko) [reply src/tree_schema.h]: Removed.

**Comment by michalvasko** [comment]
It is `lys_set_implemented` that now actually enables the features.

**Review by rkrejci — APPROVED** [review]

**Pull request merged**